File: recipe-management-domain/src/main/java/fr/cta/recipe/management/domain/repository/RecipeRepository.java
package fr.cta.recipe.management.domain.repository;

import fr.cta.recipe.management.domain.entity.Recipe;

import java.util.*;

public interface RecipeRepository {

    Optional<Recipe> getById(UUID recipeId);

    void update(Recipe recipe);

    void deleteById(UUID recipeId);

    Set<Recipe> findByOwnerId(UUID recipeOwnerId);

    void addRecipeToOwner(Recipe recipe, UUID id);
}

File: recipe-management-domain/src/main/kotlin/fr/cta/recipe/management/domain/repository/RecipeRepository.kt
package fr.cta.recipe.management.domain.repository

import com.benasher44.uuid.Uuid
import fr.cta.recipe.management.domain.entity.Recipe

interface RecipeRepository {

    fun getById(recipeId: Uuid): Recipe

    fun insert(recipe: Recipe): Unit

    fun update(recipe: Recipe): Unit

    fun deleteById(recipeId: Uuid): Unit
}